packages/gatsby/cache-dir/page-renderer.js

```diff
@@ -1,35 +1,115 @@
-import React, { createElement } from "react"
+/* global HAS_REACT_18 */
+import React, { createElement, useEffect } from "react"
 import PropTypes from "prop-types"
 import { apiRunner } from "./api-runner-browser"
 import { grabMatchParams } from "./find-path"
+import { StaticQueryContext } from "gatsby"
+import { VALID_NODE_NAMES } from "./head/constants"
+
+// Calls callback in an effect and renders children
+function Caller({ children, callback }) {
+  useEffect(() => {
+    callback()
+  }, [callback])
+
+  return children
+}
+
+// TODO: We do something similar in production-app, we  should use a take this function, put it in some utils file and share
+function render(el, Component) {
+  if (HAS_REACT_18) {
+    const reactDomClient = require(`react-dom/client`)
+    reactDomClient.createRoot(el).render(Component)
+  } else {
+    const reactDomClient = require(`react-dom`)
+    reactDomClient.render(Component, el)
+  }
+}
 
 // Renders page
-class PageRenderer extends React.Component {
-  render() {
-    const props = {
-      ...this.props,
-      params: {
-        ...grabMatchParams(this.props.location.pathname),
-        ...this.props.pageResources.json.pageContext.__params,
-      },
-    }
+function PageRenderer(props) {
+  const _props = {
+    ...props,
+    params: {
+      ...grabMatchParams(props.location.pathname),
+      ...props.pageResources.json.pageContext.__params,
+    },
+  }
+
+  const preferDefault = m => (m && m.default) || m
 
-    const pageElement = createElement(this.props.pageResources.component, {
+  const pageElement = createElement(
+    preferDefault(props.pageResources.component),
+    {
       ...props,
-      key: this.props.path || this.props.pageResources.page.path,
-    })
-
-    const wrappedPage = apiRunner(
-      `wrapPageElement`,
-      { element: pageElement, props },
-      pageElement,
-      ({ result }) => {
-        return { element: result, props }
+      key: props.path || props.pageResources.page.path,
+    }
+  )
+
+  const pageComponent = props.pageResources.component
+
+  if (pageComponent.head) {
+    if (typeof pageComponent.head !== `function`)
```

Comment:
Allow check to support class components

Reply:
Check is all fine as class components work right now. We only need to update code so we don't show a warning for class head export